src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,370 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Reflection;
+using System.Reflection.Emit;
+
+// This HttpHandlerDiagnosticListener class is applicable only for .NET 4.5/4.6, and not for .NET core.
+// If you are making these changes, please test your changes manually via custom test applications.
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// A HttpHandlerDiagnosticListener is a DiagnosticListener for .NET 4.5/4.6 where HttpClient
+    /// doesn't have a DiagnosticListener built-in. This class is not used for .NET Core because
+    /// HttpClient in .NET Core already emits DiagnosticSource events. This class compensates for
+    /// that in .NET 4.5/4.6. HttpHandlerDiagnosticListener has no public constructor, but it has a
+    /// singleton style Initialize method to control its creation.
+    /// 
+    /// To use this, the application just needs to call HttpHandlerDiagnosticListener.Initialize(), and this
+    /// will register itself with the DiagnosticListener's all listeners collection.
+    /// </summary>
+    internal class HttpHandlerDiagnosticListener : DiagnosticListener
+    {
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        {
+            Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        {
+            Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
```

Comment:
The overload above IS virtual (you have to be virtual to override an interface method).   It is simply a feature of C# that you don't have to be explicit about it.    If you add the method (you don't need override) it will do what we want (which is override anyone calling the interface).